### DISCUSSION TITLE
How do I run the benchmarks?

### DISCUSSION THREAD

**Discussion by powerseed** [RomanIakovlev/timeshape#96] [Q&A] [answered]
Could someone teach me how to run the following benchmark using sbt please?

https://github.com/RomanIakovlev/timeshape/blob/master/benchmarks/src/main/java/net/iakovlev/timeshape/BasicGeoOperationsBenchmark.java

I installed sbt but don't how what command to use. 

Thanks in advance!

**Comment by RomanIakovlev** [comment, marked as answer]
Hello there, and thanks for your interest in Timeshape! We use sbt-jmh plugin to run benchmarks. You can read about it here: https://github.com/sbt/sbt-jmh.

If you just want to quickly run the benchmarks in that class you've mentioned, here's how to do that:

1. Launch sbt console using `sbt` command from your terminal.
2. After sbt console finishes loading, switch to `benchmarks` sub-project using `project benchmarks` command.
3. Run the benchmark: `jmh:run -i 3 -wi 3 -f1 -t1 BasicGeoOperationsBenchmark`.

I recommend reading about the sbt-jmh settings to understand what the parameters mean, and how to tune them for most reliable benchmark results.

**Reply by powerseed** [reply]
Thanks for the reply, and thanks for developing this useful lib!
